### PR TITLE
Add keystore.type, deployment.name and deployment.slot settings for discovery

### PR DESCRIPTION
**Pull request by dadoonet**
In addition to PR #61, #62 and #63, this version adds support for two new options:  

* `deployment.name`: deployment name if any. Defaults to the value set with `cloud.azure.cloud.service.name`. 
* `deployment.slot`: either `staging` or `production` (default).

### CODE REVIEW

**Comment by dadoonet** [comment]
@imotov I did some changes and force pushed the new changes.
I just tested it on Azure platform and discovery works well with either old settings (deprecated) and new settings. Discovery now uses Azure SDK! \o/

Let me know what you think.

File: src/main/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProvider.java [hunk 1]
```diff
@@ -71,11 +111,43 @@ public AzureUnicastHostsProvider(Settings settings, AzureComputeService azureCom
...
-        this.host_type = HostType.valueOf(componentSettings.get(AzureComputeService.Fields.HOST_TYPE,
-                settings.get("cloud.azure." + AzureComputeService.Fields.HOST_TYPE, HostType.PRIVATE_IP.name())).toUpperCase());
+        this.refreshInterval = componentSettings.getAsTime(Fields.REFRESH,
+                settings.getAsTime("cloud.azure." + Fields.REFRESH, TimeValue.timeValueSeconds(0)));
+
+        HostType tmpHostType;
+        String strHostType = componentSettings.get(Fields.HOST_TYPE,
+                settings.get("cloud.azure." + Fields.HOST_TYPE_DEPRECATED, HostType.PRIVATE_IP.name())).toUpperCase();
+        try {
+            tmpHostType = HostType.valueOf(strHostType);
```
Comment (imotov) [inline src/main/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProvider.java]: Shouldn't it be `tmpHostType = HostType.fromString(strHostType)` instead?

File: src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    protected void doStart() throws ElasticsearchException {
+    }
+
+    @Override
+    protected void doStop() throws ElasticsearchException {
+    }
+
+    @Override
+    protected void doClose() throws ElasticsearchException {
+    }
```
Comment (imotov) [inline src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeServiceImpl.java]: Should computeManagementClient be closed here?

**Comment by imotov** [comment]
@dadoonet I left a couple of comments. Otherwise looks good to me. However, I am not an Azure expert. So, might be a good idea for somebody else to take a look.

File: src/main/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProvider.java (hunk 1 above)
Comment (dadoonet) [inline src/main/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProvider.java]: +1 good catch

File: src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeServiceImpl.java (hunk 2 above)
Comment (dadoonet) [inline src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeServiceImpl.java]: Agreed.

**Pull request merged**